cache.php

```diff
@@ -95,6 +95,26 @@ function human_size( $s ) {
 		return $s . ' ' . $size;
 	}
 
+	function machine_size( $val ) {
+        $val = trim($val);
+        $last = strtolower($val[strlen($val)-1]);
+
+        if (!is_numeric($last)){
+            $val = (int) substr($val, 0, -1);
+        }
+
+        switch($last) {
```

Comment:
Would you mind changing this into a few `if` statements instead of the `switch`?